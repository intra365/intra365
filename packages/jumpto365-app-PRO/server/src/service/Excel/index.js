module.exports.WhichToolWhen = require("./WhichToolWhen")