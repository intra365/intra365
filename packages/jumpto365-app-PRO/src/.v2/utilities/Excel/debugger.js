var Excel = require(".")
var path = require("path")
var fs = require("fs-extra")
var json = require("format-json")
Excel.parsePTO(path.join(  __dirname ,"pto-en.xlsx"),"of Office365")
.then(data => {
    fs.writeFileSync(path.join(__dirname,"pto.json"),json.plain(data))
    console.log(data)
})
.catch(err=>{
    console.log(err)

})
