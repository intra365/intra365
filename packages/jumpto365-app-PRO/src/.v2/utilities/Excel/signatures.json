[{
    "table": "skills",
    "label" : "List of Skills",
    "signature": ["Skill", "Capability"],
    "key": ["title"],
    "title": "Skill",
    "text1": "Capability"
},
{
    "table": "capabilities",
    "label" : "List of Capabilities",
    "signature": ["Capability"]

},
{
    "table": "Learning Paths",
    "label" : "List of Learning Paths",
    "signature": ["Level", "Role", "Skill"]
},

{
    "table": "Licenses",
    "label" : "List of Licenses",
    "signature": ["SKU"]
},
{
    "table": "Stories",
    "label" : "List of Stories",
    "signature": ["Story"]
},
{
    "table": "Roles",
    "label" : "List of Roles",
    "signature": ["Rolename"]
},
{
    "table": "Triggers",
    "label" : "List of Triggers",
    "signature": ["Trigger Name"]
},
{
    "table": "WTW",
    "label" : "Which tool when",
    "loader": "wtw",
    "signature": ["Area", "Use case"]
},
{
    "table": "tools",
    "label" : "List of Tools",
    "loader": "tools",
    "signature": ["key", "Title"],
    "key": ["title"],
    "title": "key",
    "text1": "Title",
    "text2": "Subtitle",
    "text3": "background",
    "text4": "path"
},
{
    "table": "tableof",
    "label" : "Periodic Table",
    "loader": "pto",
    "signature": ["", "", "1", "", "", "2"]
},
{
    "table": "groups",
    "label" : "List of Groups",
    "loader": "groups",
    "signature": ["key", "Is Connector"],
    "key": ["title"],
    "title": "key",
    "int1": "Is Connector",
    "text1": "Title",
    "text2": "Color",
    "text3": "Border Color"
}


]