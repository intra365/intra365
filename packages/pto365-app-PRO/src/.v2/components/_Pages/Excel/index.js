import React, { Component } from "react";
import PropTypes from "prop-types";
import PageLayoutPublisher from "../../_Layouts/PageLayoutPublisher";
import PageHeader from "../../PageHeader";
import FileUpload from "../../FileUpload";
import ReactJson from "react-json-view";
import {
  Dialog,
  DialogType,
  DialogFooter
} from "office-ui-fabric-react/lib/Dialog";
import Jumpto365App from "../../_Contexts/Jumpto365App";
import $ from "jquery";
import { __positioningTestPackage } from "office-ui-fabric-react/lib/utilities/positioning";
import { getExcel } from "../../../services/OfficeGraph";
import Excel from "../../../utilities/Excel";
import { navigate } from "@reach/router";
import {
  MessageBar,
  MessageBarType
} from "office-ui-fabric-react/lib/MessageBar";
import { MessageBarButton } from "office-ui-fabric-react/lib/Button";
import { ProgressIndicator } from "office-ui-fabric-react";
import background from "../../../media/Upload Page Graphic 2.png";

const storagename = "jumpto365excel";

var parseSheet = (excel, sheetName) => {
  // debugger
  return new Promise((resolve, reject) => {
    if (!excel) return reject("no excel data");
    if (!sheetName) return reject("no sheetname");

    var format = Excel.detectFormat(excel, sheetName);
    switch (format) {
      case "TOOLS":
        Excel.parseTools(excel, sheetName)
          .then(tools => {
            resolve({ format, data: tools, sheetName });
          })
          .catch(err => {
            reject(err);
          });
        break;
      case "GROUPS":
        Excel.parseGroups(excel, sheetName)
          .then(groups => {
            resolve({ format, data: groups, sheetName });
          })
          .catch(err => {
            reject(err);
          });
        break;
      case "WTW":
        Excel.parseWTW(excel, sheetName)
          .then(wtw => {
            resolve({ format, data: wtw, sheetName });
          })
          .catch(err => {
            reject(err);
          });
        break;
      case "PTO":
        Excel.parsePTO(excel, sheetName)
          .then(pto => {
            resolve({ format, data: pto, sheetName });
          })
          .catch(err => {
            reject(err);
          });
        break;
      default:
        var data = Excel.sheetToJSON(excel, sheetName);
        resolve({ format: "unknown", data, sheetName });

        break;
    }
  });
};
class Warning extends Component {
  render() {
    return this.props.message;
  }
}
/**
 * Describe overall purpose of the component
 *
 * @export
 * @class LoginPage
 * @extends {Component}
 */
export default class ExcelPage extends Component {
  static propTypes = {
    globalContext: PropTypes.obj
  };

  constructor(props) {
    super(props);
    this.state = {
      showError: false,
      sheet: null,
      type: null,
      hideDialog: true,
      files: []
    };
  }
  _closeDialog = () => {
    this.setState({ hideDialog: true });
  };

  _load = () => {
    var that = this;
    if (!this.props.filename) return;

    getExcel(this.props.filename)
      .then(excel => {
        console.log(document);
        that.setState({
          data: [
            { data: excel.data2, key: 0, file: { name: that.props.filename } }
          ]
        });
        var buf = Buffer.from(excel.data2, "base64");
        this.setState({
          excelData: buf,
          selectedFile: 0,
          id: 0,
          type: null,
          sheet: null
        });

        parseSheet(buf, that.props.sheet)
          .then(data => {
            that.setState({
              sheet: data.data,
              type: data.format,
              sheetname: that.props.sheet
            });
          })
          .catch(err => {
            that.setState({
              error: Jumpto365App.emitError(that, err, "#PARSER-GROUPS")
            });
          });
      })

      .catch(e => {
        console.log(e);
        that.setState({
          errors: (that.state.errors ? that.state.errors : []).push(e.message),
          initState: 0
        });
      });
  };
  componentDidMount = () => {
    var data = null; //sessionStorage.getItem(storagename)
    this.setState({ id: -1, data: data ? JSON.parse(data) : [] });
    window.addEventListener("resize", this.updateDimensions);
    this.updateDimensions();
    this._load();
  };

  componentDidCatch(error, info) {
    // Display fallback UI
    this.setState({
      error: Jumpto365App.emitError(this, error, "SHARING PAGE")
    });
    // You can also log the error to an error reporting service
    //logErrorToMyService(error, info);
  }

  updateDimensions = () => {
    //this.setState({ });
    var win = $(window);
    var h = win.height();
    var mastHead = $("#header");
    var pageHead = $("#pageheader");
    var toolbar = $("#toolbar");
    this.setState({
      //title: "",
      height:
        h -
        (mastHead ? mastHead.height() : 0) -
        (pageHead ? pageHead.height() : 0) -
        (toolbar ? toolbar.height() : 0),
      width: pageHead ? pageHead.width() / 2.2 : 400
    });
  };

  componentWillUnmount = () => {
    window.removeEventListener("resize", this.updateDimensions);
  };

  selectFile = selection => {
    if (selection.length === 0) return;
    if (!this.state.data) return;
    var ix = selection[0].key;

    try {
      var buf = Buffer.from(this.state.data[ix].data, "base64");
      this.setState({
        excelData: buf,
        selectedFile: ix,
        id: ix,
        type: null,
        sheet: null
      });
    } catch (error) {
      this.setState({
        error: Jumpto365App.emitError(this, error, "SELECT FILE")
      });
    }
  };

  onDrop = files => {
    this.setState({ files });
  };
  /**
   * Required method return the output of the component
   *
   * @returns
   * @memberof ScenarioPage
   */
  render() {
    var that = this;
    let storedFiles = this.state.data
      ? this.state.data.map(function(item, ix) {
          return { key: item.key, name: item.file.name };
        })
      : [];

    var context = this.props.globalContext;
    function prompt(id) {
      return Jumpto365App.prompt(id, context);
    }

    return (
      <PageLayoutPublisher>
        <PageHeader
          title={
            prompt("42") + " - drop a file or click to upload ..." /* Upload */
          }
          color="#2a7ab9"
        />

        <div>
          {this.state.errors && (
            <ReactJson collapsed="0" src={this.state.errors} />
          )}
          {this.state.showError && (
            <div style={{ marginRight: "-16px", marginTop: "-8px" }}>
              {" "}
              <MessageBar
                className="messagedialog"
                messageBarType={MessageBarType.error}
                isMultiline={false}
                actions={null}
                onDismiss={() => {
                  that.setState({ showError: false });
                }}
                dismissButtonAriaLabel="Dismiss"
              >
                {this.state.error}
              </MessageBar>
            </div>
          )}
          <div style={{ display: "flex" }}>
            <div
              style={{
                flexGrow:2,
                backgroundColor: "#c3c3c3",
                marginTop: "-8px",
                backgroundPosition: "center",
                backgroundRepeat: "no-repeat",
                backgroundImage: `url(${background})`
              }}
            >
              <FileUpload
                label="Upload"
                onUpload={(data, file) => {
                  try {
                    switch (file.type) {
                      case "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet":
                        var buf = Buffer.from(data, "base64");
                        this.state.data.push({
                          key: this.state.data.length,
                          file: {
                            name: file.name,
                            size: file.size,
                            type: file.type,
                            lastModified: file.lastModified
                          },
                          data
                        });
                        sessionStorage.setItem(
                          storagename,
                          JSON.stringify(this.state.data)
                        );
                        navigate(
                          `/excel/local/${this.state.data.length}/of Office365`
                        );
                        this.setState({
                          data: this.state.data,
                          selectedFile: sessionStorage.length,
                          excelData: buf,
                          file: { filename: file.name, size: file.size }
                        });
                        break;
                      case "application/vnd.openxmlformats-officedocument.wordprocessingml.document":
                        var buf = Buffer.from(data, "base64");
                        this.state.data.push({
                          key: this.state.data.length,
                          file: {
                            name: file.name,
                            size: file.size,
                            type: file.type,
                            lastModified: file.lastModified
                          },
                          data
                        });
                        sessionStorage.setItem(
                          storagename,
                          JSON.stringify(this.state.data)
                        );
                        navigate(`/word/local/${this.state.data.length}`);
                        this.setState({
                          data: this.state.data,
                          selectedFile: sessionStorage.length,
                          excelData: buf,
                          file: { filename: file.name, size: file.size }
                        });
                        break;
                      case "image/jpeg":
                        debugger;
                        var buf = Buffer.from(data, "base64");

                        this.state.data.push({
                          key: this.state.data.length,
                          file: {
                            name: file.name,
                            size: file.size,
                            type: file.type,
                            lastModified: file.lastModified
                          },
                          data
                        });

                        break;
                      default:
                        this.setState({
                          showError: true,
                          error: `${prompt("44")} "${file.type}" ${prompt(
                            "45"
                          )} `
                        });

                        break;
                    }
                  } catch (error) {
                    this.setState({ showError: true, error: error.message });
                  }
                }}
              />
            </div>
            <div
              style={{
               
                padding: "16px",
              
                flexGrow:1,
              }}
            >
              <h3>Templates</h3>
              <p>
              Grab one of the templates to get started
              </p>
              <div>
                  
                  Word
                <ul>
                    
                    <li>
                    <a href="https://github.com/jumpto365/microsoft365/raw/master/templates/New%20Article.docx">
                     New Article
                    </a>
                    </li>
                  <li>                   
                    
                    <a href="https://github.com/jumpto365/microsoft365/raw/master/templates/Periodic%20Table%20of%20Office%20365.docx">
                      All articles used in the Periodic Table of Office 365 in
                      english
                    </a>
                  </li>
                </ul>
              </div>
              <div>
                {" "}
                Excel
                <ul>
                    
                  <li>
                    <a href="https://github.com/jumpto365/microsoft365/raw/master/templates/Customised%20Periodic%20Table%20of.xlsx">
                      Periodic Table of Office 365 - English version
                    </a>
                  </li>
                </ul>{" "}
              </div>
            </div>
          </div>

          {/* <ReactJson collapsed="2" src={this.state} />  */}
        </div>
      </PageLayoutPublisher>
    );
  }
}
