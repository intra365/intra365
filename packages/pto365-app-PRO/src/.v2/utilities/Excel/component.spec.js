import React, { Component } from 'react';

import Excel from '.'

test('Check that it can detect WTW', () => {


    
    const format  =  Excel.detectFormat(mocks.excelData(),"for Meetings")

    expect(format).toBe("WTW");
 
}

)
test('Check that it can detect PTO', () => {


    
    const format  =  Excel.detectFormat(mocks.excelData(),"of Office365")

    expect(format).toBe("PTO");
 
}

)



test('Check that it renders', () => {




    const wrapper =  Excel.toJSON(mocks.excelData())

    expect(wrapper).toMatchSnapshot();
 
}

)