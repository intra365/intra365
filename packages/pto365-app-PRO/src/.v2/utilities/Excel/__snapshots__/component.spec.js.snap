// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that it renders 1`] = `
Object {
  "sheets": Array [
    Object {
      "data": Array [],
      "name": "About",
    },
    Object {
      "data": Array [
        Object {
          "the PERIODIC TABLE of": "YOUR CONTEXT",
        },
        Object {
          "__EMPTY_1": "Author",
          "__EMPTY_5": "Niels Gregers Johansen",
        },
        Object {
          "__EMPTY_1": "Email",
          "__EMPTY_5": "niels@jumpto365.com",
        },
        Object {
          "__EMPTY_1": "License",
          "__EMPTY_5": "MIT",
        },
        Object {
          "__EMPTY_1": "Revision",
          "__EMPTY_5": "A1",
        },
      ],
      "name": "of Template",
    },
    Object {
      "data": Array [
        Object {
          "Which tool When": "MEETING PREPERATIONS",
          "__EMPTY_7": "Author",
          "__EMPTY_8": "Credits",
          "__EMPTY_9": "License",
        },
        Object {
          "__EMPTY_7": "Niels Gregers Johansen",
        },
        Object {
          "Which tool When": "Area",
          "__EMPTY": "Task",
          "__EMPTY_1": "Best",
          "__EMPTY_10": "da",
          "__EMPTY_2": "Better ",
          "__EMPTY_3": "Worse",
          "__EMPTY_4": "Worst",
          "__EMPTY_5": "No go",
          "__EMPTY_6": "Tags",
          "__EMPTY_7": "Preview",
          "__EMPTY_8": "Url",
          "__EMPTY_9": "en",
        },
        Object {
          "Which tool When": "Prepare",
          "__EMPTY_9": "Collaboration",
        },
        Object {
          "__EMPTY": "Collaborate on creating a memo with someone else",
          "__EMPTY_1": "OneDrive",
          "__EMPTY_10": "Rediger et dokument samtidig med en anden inden publisering",
          "__EMPTY_6": "File Storage & Collaboration",
          "__EMPTY_7": "https://static.makeuseof.com/wp-content/uploads/2015/11/manage-shared-files-drive-670x335.jpg",
          "__EMPTY_8": "https://www.dr.dk",
        },
        Object {
          "__EMPTY": "Collaborate on creating a agenda with someone else",
          "__EMPTY_1": "OneDrive",
          "__EMPTY_10": "Rediger et dokument samtidig med en anden inden publisering",
          "__EMPTY_6": "File Storage & Collaboration",
          "__EMPTY_7": "https://st2.depositphotos.com/6741230/11510/v/950/depositphotos_115106712-stock-illustration-agenda-list-icon-vector-illustration.jpg",
          "__EMPTY_8": "https://www.dr.dk",
        },
        Object {
          "__EMPTY": "Save and Edit Personal document",
          "__EMPTY_1": "OneDrive",
          "__EMPTY_10": "Gem og rediger personlige dokumenter",
          "__EMPTY_4": "Outlook",
          "__EMPTY_7": "https://fthmb.tqn.com/2Mb91JHmRFuWlYNoKk0xy5IyrwE=/768x0/filters:no_upscale():max_bytes(150000):strip_icc()/iStock-596780340-edit-word-free-58c8489c5f9b58af5c2ee4c1.jpg",
        },
        Object {
          "__EMPTY": "Make an anonymous survey",
          "__EMPTY_10": "Stil et spørgsmål uden at vide, hvem der har svaret",
          "__EMPTY_2": "Tavlen",
          "__EMPTY_3": "skype",
          "__EMPTY_4": "Outlook",
          "__EMPTY_7": "http://www.sogosurvey.com/blog/wp-content/uploads/2016/02/When-to-Make-a-Survey-Anonymous-1024x638.jpg",
        },
        Object {
          "__EMPTY": "Setup a site for long lasting dialogue",
          "__EMPTY_1": "Tavlen",
          "__EMPTY_10": "Opret en debat om et emne med forventet længere levetid",
          "__EMPTY_4": "Projekt,Outlook",
          "__EMPTY_7": "https://www.procurious.com/blog-content/2015/12/Discussion-Forum.jpg",
        },
        Object {
          "__EMPTY": "Make a poll for a specifc group of persons",
          "__EMPTY_1": "Afstemning",
          "__EMPTY_10": "Opret en afstemning eller undersøgelse for en bestemt gruppe",
          "__EMPTY_3": "Projekt",
          "__EMPTY_7": "http://www.afd-techtalk.com/wp-content/uploads/2018/02/poll-time.jpg",
        },
        Object {
          "__EMPTY": "Send a document for proof reading",
          "__EMPTY_1": "Projekt",
          "__EMPTY_10": "Send et dokument til korrektur",
          "__EMPTY_3": "onedrive",
          "__EMPTY_7": "https://juliaproofreader.files.wordpress.com/2014/04/whizz.jpg",
        },
        Object {
          "__EMPTY_3": "outlook",
        },
        Object {
          "__EMPTY": "Invite a colleague for lunch",
          "__EMPTY_1": "Skype",
          "__EMPTY_10": "Inviter en kollega til frokost",
          "__EMPTY_3": "Outlook",
          "__EMPTY_7": "http://st4.styapokupayu.ru/images/blog_posts/covers/000/157/712_large.jpg?1456488984",
        },
        Object {
          "__EMPTY": "Create an interactive and creative presentation",
          "__EMPTY_1": "Sway",
          "__EMPTY_10": "Skab en interaktiv og kreativ præsentation til et møde",
          "__EMPTY_7": "http://focusky.com/learning-center/wp-content/uploads/2017/04/creative-presentation-ideas-to-create-extraordinary-presentation-2.png",
        },
        Object {
          "__EMPTY": "Invite specific persons to a meeting",
          "__EMPTY_1": "Outlook",
          "__EMPTY_10": "Inviter udvalgte personer til et møde",
          "__EMPTY_2": "Teams",
          "__EMPTY_7": "http://1.bp.blogspot.com/-S3fbBqWwuPs/Th8R4MdzZZI/AAAAAAAABb8/nd1hr8Grto0/s1600/meeting.gif",
        },
        Object {
          "__EMPTY": "Self serviced booking of you or your team",
          "__EMPTY_1": "Booking",
          "__EMPTY_10": "Selvbetjent booking af dig eller dit team",
          "__EMPTY_2": "Outlook",
          "__EMPTY_7": "https://blog.one.com/wp-content/uploads/2016/02/teambooking.png",
        },
        Object {
          "Which tool When": "Meet",
          "__EMPTY_9": "Communication",
        },
        Object {
          "__EMPTY": "Deltag i en chat med en kollega/samarbejdspartner",
          "__EMPTY_10": "Deltag i en chat med en kollega/samarbejdspartner",
          "__EMPTY_2": "Newsfeed",
          "__EMPTY_4": "Outlook",
          "__EMPTY_6": "Informal",
          "__EMPTY_7": "https://img.freepik.com/free-vector/messenger-application-to-chat-in-flat-style_23-2147804300.jpg?size=338&ext=jpg",
        },
        Object {
          "__EMPTY": "Conduct an online-meeting with your team or a colleague",
          "__EMPTY_1": "Skype",
          "__EMPTY_10": "Afhold et online-møde med dit team eller en kollega",
          "__EMPTY_7": "https://www.videonations.co.uk/wp-content/uploads/2015/01/Meeting-room-video-collaboration.jpg",
        },
        Object {
          "__EMPTY": "Conduct an online-meeting with external",
          "__EMPTY_1": "Skype",
          "__EMPTY_10": "Afhold et online-møde med eksterne",
          "__EMPTY_2": "Teams",
          "__EMPTY_3": "Skype (personal)",
          "__EMPTY_7": "https://www.videonations.co.uk/wp-content/uploads/2015/01/Meeting-room-video-collaboration.jpg",
        },
        Object {
          "__EMPTY_2": "Goto Meeting",
          "__EMPTY_7": "https://www.videonations.co.uk/wp-content/uploads/2015/01/Meeting-room-video-collaboration.jpg",
        },
        Object {
          "__EMPTY": "Make notes (shareable as well as individual) from a meeting",
          "__EMPTY_1": "OneNote",
          "__EMPTY_10": "Tag noter (fælles såvel som individuelt) fra et møde",
          "__EMPTY_4": "Outlook",
          "__EMPTY_7": "https://cdn.lynda.com/course/373782/373782-636247511734821922-16x9.jpg",
        },
        Object {
          "__EMPTY": "Assign task to specific persons",
          "__EMPTY_1": "Planner",
          "__EMPTY_10": "Tildel opgaver til specifikke personer ",
          "__EMPTY_3": "Kanban",
          "__EMPTY_4": "Project",
          "__EMPTY_7": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRRJC1JKuF6LNQcvq52GwxrfPiDxGHL9luGLiiLVsWls_2zRNOq",
        },
        Object {
          "__EMPTY": "Assign task to general roles",
          "__EMPTY_1": "Projekt",
          "__EMPTY_10": "Tildel opgaver til generelle roller",
          "__EMPTY_7": "http://creativeskillset.org/assets/0002/1683/Art_Roles_VFX_large.png",
        },
        Object {
          "__EMPTY": "Vær kreativ",
          "__EMPTY_2": "Tegne",
          "__EMPTY_7": "http://www.kb.dk/export/sites/kb_dk/da/Billedgalleri/KULT/Udstillinger/winding_28_1000px.jpg",
        },
      ],
      "name": "for Meetings",
    },
    Object {
      "data": Array [
        Object {
          "1": "dynamics365",
          "8": "sway",
          "__EMPTY": "1",
          "__EMPTY_16": "Presentations",
        },
        Object {
          "1": "stream",
          "2": "visio",
          "5": "word",
          "6": "excel",
          "7": "onenote",
          "8": "powerpoint",
          "__EMPTY": "2",
          "__EMPTY_16": "Office Online",
        },
        Object {
          "3": "Employee\\\\nProfiles",
        },
        Object {
          "1": "sharepoint",
          "2": "onedrive",
          "3": "delve",
          "5": "forms",
          "6": "powerapps",
          "7": "flow",
          "8": "powerbi",
          "__EMPTY": "3",
          "__EMPTY_1": "File Storage & Collaboration",
          "__EMPTY_16": "Business App Platform",
          "__EMPTY_9": "Forms",
        },
        Object {
          "1": "mail",
          "2": "calendar",
          "3": "people",
          "4": "tasks",
          "6": "todo",
          "7": "planner",
          "8": "project",
          "__EMPTY": "4",
          "__EMPTY_1": "Outlook",
          "__EMPTY_11": "Task\\\\nManagement",
          "__EMPTY_16": "Project Management",
          "__EMPTY_2": "Outlook",
          "__EMPTY_3": "Outlook",
          "__EMPTY_4": "Outlook",
          "__EMPTY_5": "Outlook",
          "__EMPTY_6": "Outlook",
          "__EMPTY_7": "Outlook",
        },
        Object {
          "1": "skype",
          "2": "teams",
          "3": "ServiceNow",
          "4": "yammer",
          "7": "bookings",
          "8": "staffhub",
          "__EMPTY": "5",
          "__EMPTY_1": "Chat & Conferencing",
          "__EMPTY_11": "Social\\\\nNetworking",
          "__EMPTY_16": "Small Business Applications",
        },
        Object {
          "1": "Direct Communication",
        },
      ],
      "name": "of Office365",
    },
    Object {
      "data": Array [
        Object {
          "Tags": "red",
        },
        Object {
          "Tags": "green",
        },
        Object {
          "Tags": "yellow",
        },
      ],
      "name": "tags",
    },
    Object {
      "data": Array [
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Bookings",
          "Name": "bookings",
          "color": "#25abb2",
          "icon": "https://jumpto365.com/resources/images/app/Bookings.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Calendar",
          "Link": "https://outlook.office.com/owa/?path=/calendar/view/Day",
          "Name": "calendar",
          "color": "#2072b9",
          "icon": "https://jumpto365.com/resources/images/app/Calendar.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Delve",
          "Name": "delve",
          "color": "#1c7bb9",
          "icon": "https://jumpto365.com/resources/images/app/Delve.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Dynamics",
          "Name": "dynamics365",
          "color": "#15224e",
          "icon": "https://jumpto365.com/resources/images/app/Dynamics.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Excel",
          "Link": "https://office.live.com/start/Excel.aspx",
          "Name": "excel",
          "color": "#02723b",
          "icon": "https://jumpto365.com/resources/images/app/Excel.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Flow",
          "Link": "https://flow.microsoft.com/en-us/",
          "Name": "flow",
          "color": "#0077ff",
          "icon": "https://jumpto365.com/resources/images/app/Flow.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Forms",
          "Link": "https://forms.office.com/",
          "Name": "forms",
          "color": "#008271",
          "icon": "https://jumpto365.com/resources/images/app/Forms.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Mail",
          "Link": "https://outlook.office.com/owa/",
          "Name": "mail",
          "color": "#2072b9",
          "icon": "https://jumpto365.com/resources/images/app/Mail.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Newsfeed",
          "Name": "newsfeed",
          "color": "#0072c6",
          "icon": "https://jumpto365.com/resources/images/app/Newsfeed.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/OneDrive",
          "Name": "onedrive",
          "color": "#1174c3",
          "icon": "https://jumpto365.com/resources/images/app/OneDrive.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/OneNote",
          "Link": "https://www.onenote.com/notebooks",
          "Name": "onenote",
          "color": "#80397b",
          "icon": "https://jumpto365.com/resources/images/app/OneNote.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/People",
          "Link": "https://outlook.office.com/owa/?path=/people",
          "Name": "people",
          "color": "#2072b9",
          "icon": "https://jumpto365.com/resources/images/app/People.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Planner",
          "Link": "https://tasks.office.com/",
          "Name": "planner",
          "color": "#3c853c",
          "icon": "https://jumpto365.com/resources/images/app/Planner.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/PowerApps",
          "Link": "https://web.powerapps.com/home",
          "Name": "powerapps",
          "color": "#742774",
          "icon": "https://jumpto365.com/resources/images/app/PowerApps.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Power-BI",
          "Link": "https://app.powerbi.com/",
          "Name": "powerbi",
          "color": "#f2c811",
          "icon": "https://jumpto365.com/resources/images/app/Power-BI.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/PowerPoint",
          "Link": "https://office.live.com/start/PowerPoint.aspx",
          "Name": "powerpoint",
          "color": "#d24726",
          "icon": "https://jumpto365.com/resources/images/app/PowerPoint.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Project",
          "Link": "http://www.microsoftproject-online.com/",
          "Name": "project",
          "color": "#2f722d",
          "icon": "https://jumpto365.com/resources/images/app/Project.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/SharePoint",
          "Name": "sharepoint",
          "color": "#0072c6",
          "icon": "https://jumpto365.com/resources/images/app/SharePoint.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Skype",
          "Name": "skype",
          "color": "#00aff0",
          "icon": "https://jumpto365.com/resources/images/app/Skype.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/StaffHub",
          "Name": "staffhub",
          "color": "#25c8c2",
          "icon": "https://jumpto365.com/resources/images/app/StaffHub.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Stream",
          "Link": "https://web.microsoftstream.com/",
          "Name": "stream",
          "color": "#c30052",
          "icon": "https://jumpto365.com/resources/images/app/Stream.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Sway",
          "Name": "sway",
          "color": "#008272",
          "icon": "https://jumpto365.com/resources/images/app/Sway.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Tasks",
          "Link": "https://outlook.office.com/owa/?path=/tasks",
          "Name": "tasks",
          "color": "#2072b9",
          "icon": "https://jumpto365.com/resources/images/app/Tasks.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Teams",
          "Link": "https://teams.microsoft.com/",
          "Name": "teams",
          "color": "#3f418e",
          "icon": "https://jumpto365.com/resources/images/app/Teams.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/To-Do",
          "Link": "https://todo.microsoft.com/?app",
          "Name": "todo",
          "color": "#3896f9",
          "icon": "https://jumpto365.com/resources/images/app/To-Do.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Visio",
          "Name": "visio",
          "color": "#4568c4",
          "icon": "https://jumpto365.com/resources/images/app/Visio.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Word",
          "Link": "https://office.live.com/start/Word.aspx",
          "Name": "word",
          "color": "#2b579a",
          "icon": "https://jumpto365.com/resources/images/app/Word.png",
        },
        Object {
          "Content": "https://hexatown.github.io/docs/microsoft/office365/Yammer",
          "Link": "https://www.yammer.com/",
          "Name": "yammer",
          "color": "#007dc6",
          "icon": "https://jumpto365.com/resources/images/app/Yammer.png",
        },
      ],
      "name": "tools",
    },
    Object {
      "data": Array [
        Object {
          "Name": "File Storage & Collaboration",
        },
        Object {
          "Name": "Outlook",
        },
        Object {
          "Name": "Meet",
        },
        Object {
          "Name": "Prepare",
        },
      ],
      "name": "groups",
    },
  ],
}
`;
